posthog/hogql/parser.py

```diff
@@ -23,19 +25,28 @@
     parse_select as _parse_select_cpp,
 )
 
-RULE_TO_PARSE_FUNCTION = {
+RULE_TO_PARSE_FUNCTION: Dict[Literal["python", "cpp"], Dict[Literal["expr", "order_expr", "select"], Callable]] = {
     "python": {
         "expr": lambda string, start: HogQLParseTreeConverter(start=start).visit(get_parser(string).expr()),
         "order_expr": lambda string: HogQLParseTreeConverter().visit(get_parser(string).orderExpr()),
         "select": lambda string: HogQLParseTreeConverter().visit(get_parser(string).select()),
     },
     "cpp": {
-        "expr": lambda string, _: _parse_expr_cpp(string),  # The start arg is ignored in the C++ version
+        "expr": lambda string, start: _parse_expr_cpp(string, is_internal=start is None),
         "order_expr": lambda string: _parse_order_expr_cpp(string),
         "select": lambda string: _parse_select_cpp(string),
     },
 }
 
+RULE_TO_HISTOGRAM: Dict[Literal["expr", "order_expr", "select"], Histogram] = {
+    rule: Histogram(
+        f"parse_{rule}_seconds",
+        f"Time to parse {rule} expression",
+        labelnames=["backend"],
+    )
+    for rule in RULE_TO_PARSE_FUNCTION["python"]
```

Comment:
so we're only histogramm-ing python rules?

otherwise this looks good.

the buckets are set up to measure typical web requests which might be fine here but can need tweaking once you've seen how they fall out (e.g. lots between 100-300 ms and you want to dig in to see where they lie in that range)